src/managers/config.py

```diff
@@ -256,56 +271,112 @@ def scram_properties(self) -> list[str]:
         username = INTER_BROKER_USER
         password = self.state.cluster.internal_user_credentials.get(INTER_BROKER_USER, "")
 
+        listener_name = self.internal_listener.name.lower()
+        listener_mechanism = self.internal_listener.mechanism.lower()
+
         scram_properties = [
-            f'listener.name.{self.internal_listener.name.lower()}.scram-sha-512.sasl.jaas.config=org.apache.kafka.common.security.scram.ScramLoginModule required username="{username}" password="{password}";'
-        ]
-        client_scram = [
-            auth.name for auth in self.client_listeners if auth.protocol.startswith("SASL_")
+            f'listener.name.{listener_name}.{listener_mechanism}.sasl.jaas.config=org.apache.kafka.common.security.scram.ScramLoginModule required username="{username}" password="{password}";',
+            f"listener.name.{listener_name}.sasl.enabled.mechanisms={self.internal_listener.mechanism}",
         ]
-        for name in client_scram:
-            scram_properties.append(
-                f'listener.name.{name.lower()}.scram-sha-512.sasl.jaas.config=org.apache.kafka.common.security.scram.ScramLoginModule required username="{username}" password="{password}";'
+
+        for auth in self.client_listeners:
+            if not auth.mechanism.startswith("SCRAM"):
+                continue
+            scram_cfg = textwrap.dedent(
+                f"""\
+                listener.name.{auth.name.lower()}.{auth.mechanism.lower()}.sasl.jaas.config=org.apache.kafka.common.security.scram.ScramLoginModule required username="{username}" password="{password}";
+                listener.name.{auth.name.lower()}.sasl.enabled.mechanisms={auth.mechanism}
+                """
             )
+            scram_properties.append(scram_cfg)
 
         return scram_properties
 
     @property
-    def security_protocol(self) -> AuthMechanism:
+    def oauth_properties(self) -> list[str]:
+        """Builds the properties for the oauth listener.
+
+        Returns:
+            list of oauth properties to be set.
+        """
```

Comment:
**suggestion** To avoid nesting and too many if statements, a more readable approach could be to first check whether we have a `oauth` relation, and if so, just rely on the assumption that a OAuth client exists, something like:

```
if not self.state.oauth_relation:
    return []

listener = [listener for listener in self.client_listeners if listener.mechanism.startswith("OAUTH")][0]

validation_cfg = ...

return [oauth_cfg]
```

Reply:
updated as suggested